### PR TITLE
Migrate kafkaexporter from Collector core to Collector contrib 

### PR DESCRIPTION
**Pull request by Saber-W**
**Description:**
This PR migrates the kafkaexporter component from the Collector core repository to the Collector contrib repository.

This PR retains the commit history from core and includes plumbing fixes for go.mod, makefile, components.go.

**Link to tracking Issue:**
[opentelemetry-collector/#3474](https://github.com/open-telemetry/opentelemetry-collector/issues/3474)
[opentelemetry-collector/#3649](https://github.com/open-telemetry/opentelemetry-collector/issues/3649)

### CODE REVIEW

**Comment by alolita** [comment]
Hi @bogdandrutu @tigrannajaryan based on our discussion last week, we're starting to file PRs to move each core component to contrib. Please review and merge. Thanks!

**Comment by bogdandrutu** [comment]
Please make sure you use the new translator module that you created for opencensus/jaeger if needed instead of the one in collector.

File: exporter/kafkaexporter/go.mod [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/exporter/kafkaexporter
+
+go 1.16
+
+require (
+	github.com/Shopify/sarama v1.29.1
+	github.com/gogo/protobuf v1.3.2
+	github.com/jaegertracing/jaeger v1.25.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.0.0-00010101000000-000000000000
+	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger v0.0.0-20210821065208-3703e0d1dcaa
```
Comment (Aneurysm9) [inline exporter/kafkaexporter/go.mod:10]: Should this still be referencing `00010101...` with a `replace` directive?

Comment (Saber-W) [reply exporter/kafkaexporter/go.mod:10]: Using `00010101...` would be fine for jaeger, but if we use this for opencensus, an error like `github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/opencensus@v0.0.0-00010101000000-000000000000: invalid version: unknown revision 000000000000` will occur.

Comment (bogdandrutu) [reply exporter/kafkaexporter/go.mod:10]: You need that same replace statement in the root go.mod

Comment (Saber-W) [reply exporter/kafkaexporter/go.mod]: Right, makes sense. I have made the changes.

**Review by anuraaga — APPROVED** [review]
I'd recommend writing a pinned issue describing this migration effort, what/why/how. In particular, I just noticed the approach of copying the `opentelemetry-core` repo into `internal/core` in advance to then start integrating, which is a neat trick. But documenting this for reviewer benefit and also to keep a record for future archaeology is important. There is #3474, if it's an appropriate issue for this, than I would suggest rewriting the first comment in the issue to include details and the linked issues, etc, presenting it for users. /cc @alolita 

It's unfortunate that the commit message in https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/c7136470e5caa9fb9135f2c586767516e17db677, which is what will show up in git blame for years to come, is basically empty - that one deserved a lot of detail. At the least though GitHub links it to the PR which links to #3474 so we can at least flesh that issue out.

**Pull request merged**